### DISCUSSION TITLE
Minimap that controls the main viewport

### DISCUSSION THREAD

**Discussion by mcost45** [visgl/deck.gl#5788] [Q&A] [answered]
Working with orthographic layers, I already have a minimap view set up that displays the main viewport bounds (like https://deck.gl/examples/multi-view/), but I've had some difficulty with implementing the ability to interact with the minimap - using mouse events to move the main viewport (e.g. panning on the minimap view moves the main viewport to match) I've tried a couple different ideas like a custom controller - or handling onViewStateChange seperately for the minimap view.

In the case of the latter, I have added a controller on the minimap view. The minimap view is not supposed to change states (as it is a static overview), so I do not apply any of the new viewStates to the minimap view. Instead, I've attempted to use the minimap view's new target values, and try and map these into new targets to set on the main view. However I am yet to find a perfect way to map the target from the minimap view to the main view: using the initial main view targets minus the new minimap view targets is close but still not correct.

Does any one have any suggestions for mapping target values between views, or know of a better way to do this (move the main viewport via the minimap), or have an example of such an implementation?

Thank you for any help!!

**Comment by Pessimistress** [comment, marked as answer]
You have to update the view state of a view if you want to enable controller. In your use case, you can add a third view that is identical to the minimap, but does not render anything, and set its `controller: true`.

**Comment by mcost45** [comment]
Thanks to @Pessimistress for pointing me in the right direction! In case anyone else wants to do the same:

- On main view state changes, update the 'dynamic' (controllable) minmap view target to the negative of the new main view target
- On the dynamic minimap view state changes, update its view state to match the new view state, EXCEPT: Set its target to be the negative of the new target, and do not change its zoom level; keep it the same as its initial (matching the 'static' minimap view zoom)
